networkx/drawing/nx_pylab.py

```diff
@@ -430,20 +432,41 @@ def draw_networkx_nodes(
         node_color = apply_alpha(node_color, alpha, nodelist, cmap, vmin, vmax)
         alpha = None
 
-    node_collection = ax.scatter(
-        xy[:, 0],
-        xy[:, 1],
-        s=node_size,
-        c=node_color,
-        marker=node_shape,
-        cmap=cmap,
-        vmin=vmin,
-        vmax=vmax,
-        alpha=alpha,
-        linewidths=linewidths,
-        edgecolors=edgecolors,
-        label=label,
-    )
+    if isinstance(node_shape, list):
+        for i, s in enumerate(node_shape):
+            node_collection = ax.scatter(
+                xy[i, 0],
+                xy[i, 1],
+                s=node_size[i] if isinstance(node_size, list) else node_size,
```

Comment:
There is a recurring pattern of `x[i[ if isinstance(x,list) else x`, and I wonder if it would be better to force x to be a list if it's not a list, e.g. via `x` = [x]*len(node_shape)`, and then pass all the relevant lists to `zip` (enumerate will not be needed).

This is a subjective preference, so not sure if others feel this would be more readable.

Reply:
And in the above suggestion, of course the forcing to the list would be if `x` is not already a list.

Reply:
The issue is it is valid for some to be lists or (float) values, for instance, you can the alpha can be a list (corresponding to an the individual nodes), a single value or none. This was the most elegant way I thought of bundling these cornercases.

Reply:
I agree with @dschult .

I'm -1 on handling this explicitly in `nx_pylab` as it makes the logic around argument interpretation even more opaque.

FWIW you could accomplish this with the current machinery using "nodelist", e.g. something like:

```python
G = nx.DiGraph()
nx.add_path(G, range(10))
pos = {n: (0, n) for n in G}
# Or whatever marker specifications you want
markers = itertools.cycle(["s", "o", "*", "^", "v", "<", ">", "h"])
fig, ax = plt.subplots()
for n, marker in zip(G, markers):
    color = "tab:olive" if n in {2, 5, 6, 9} else "tab:cyan"
    nx.draw_networkx_nodes(G, pos, nodelist=[n], node_color=color, node_shape=marker)
nx.draw_networkx_edges(G, pos)

Reply:
I agree @dschult that this change would be more impactful in Matplotlib---which is why I was amazed that this simple functionality didn't exist; I mean, it supports everything else to `scatter` as a list *except* node_shapes!
Unfortunately, when I looked at the `scatter` function in Matplotlib it was a bit more complex to implement and couldn't make rapid headway. Since I only need this functionality (in Networkx) for some visualizations with my job, and it was much easier to implement here, I opened the PR on the off-chance that it would be useful. If anyone else wants to shepard/advocate for a similar change into Matplotlib I'd be appreciative (but won't hold my breath). For my own purposes I'll be using my forked unincorporated change for my work tooling.

@rossbar I was unaware of the nodelist (it looks cool) but ultimately has the added complexity I was looking to avoid with this change---why require the user to iterate with draw calls when it's something the library can do internally? I know it'd make for cleaner more interpretable code with my suggested change.